src/PhpImap/Mailbox.php

```diff
@@ -589,6 +597,14 @@ public function getMail($mailId, $markAsSeen = true) {
 	}
 
 	protected function initMailPart(IncomingMail $mail, $partStructure, $partNum, $markAsSeen = true) {
+		
+		if ($this->attachmentsIgnore && 
+		($partStructure->type !== TYPEMULTIPART && 
+		($partStructure->type !== TYPETEXT || !in_array(strtolower($partStructure->subtype), ['plain','html']))))
```

Comment:
I totally agree we should avoid processing parts that we don't need but I am not sure this is the way.

That sure is a lot of ifs are we certain we have covered all the basis? Just looks like something that can cause major headaches constantly having to patch for edge cases slipping through or getting ignored, isn't this a wheel we already have? Since we are checking now is it still necessary to do `isAttachment` again?

https://github.com/barbushin/php-imap/blob/56ed03e07eea67dd8dc7306c66df093ac1a82294/src/PhpImap/Mailbox.php#L652 

Instead of something that is supposed to optimise performance we've added extra checks and balances, being punished to get the reward.